### PR TITLE
Migrates PostServiceRemotes to WordPressKit framework

### LINKED ISSUE #7320: Migrate PostServiceRemoteREST to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### LINKED ISSUE #7321: Migrate PostServiceRemoteXMLRPC to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #7320
Fixes #7321  

This PR:
* Migrates the `RemotePost.h/.m` model to the WPKit framework
* Migrates the `RemotePostCategory.h/.m` model to the WPKit framework
* Moves `PostStatus*` externs out of `AbstractPost.h` and into `RemotePost.h` (all imports in other files updated accordingly)
* Migrates the `PostServiceRemote.h` to the WPKit framework
* Migrates the `PostServiceRemoteREST.h/.m` to the WPKit framework
* Migrates the `PostServiceRemoteXMLRPC.h/.m` to the WPKit framework
* _Copies_ `XMLRPCTestable.swift` and related mocks into WPKitTests (will eventually be removed from WordPress app)
* Migrates related unit tests (`PostServiceRemoteRESTTests.m`) + mock data to WPKitTest
* Migrates related unit tests (`PostServiceRemoteXMLRPCTests.swift`) + mock data to WPKitTest
* Migrates `DisplayableImageHelper.h/.m` to the WPShared framework
* Migrates `DisplayableImageHelperTest.m` tests to WPShared framework
* Updates related imports/refs in main WordPress project

Ref #7301

**To test:**
Since a lot of code involving posts was moved into WPKit, please test this thoroughly. 😄 

1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app:
a. Verify You can view/create/update/trash/delete/restore posts on .com sites
b. Verify You can view/create/update/trash/delete/restore pages on .com sites
c. Verify You can view/create/update/trash/delete/restore posts on .org sites
d. Verify You can view/create/update/trash/delete/restore pages on .org sites

@astralbodies Would you mind reviewing? _Greatly_ appreciated!

### CODE REVIEW

**Comment by bummytime** [comment]
Also adding @elibud as a reviewer...the more, the merrier. 🎉

File: WordPress/WordPressTest/AbstractPostTest.swift [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import XCTest
```
Comment (bummytime) [inline WordPress/WordPressTest/AbstractPostTest.swift]: 💥

**Comment by bummytime** [comment]
Roger that @astralbodies...I'll take a peek at the rules.

**Comment by bummytime** [comment]
Ok, @astralbodies. That last commit made the analyzer rules the same between WPShared, WPKit, and the main WP app. Give it another go!

**Comment by astralbodies** [comment]
Good to go @bummytime - :shipit:

**Comment by bummytime** [comment]
Thanks @astralbodies!

**Pull request merged**